src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -60,7 +79,20 @@ public Project addProjectToSubmissionEnvelope(SubmissionEnvelope submissionEnvel
                                      });
     }
 
-    public Page<Project> queryByContent(List<MetadataCriteria> query, Pageable pageable){
-        return this.projectRepository.findByContent(query, pageable);
+    public Page<Project> findByCriteria(List<MetadataCriteria> criteriaList, Boolean andCriteria, Pageable pageable){
+        return this.projectRepository.findByCriteria(criteriaList, andCriteria, pageable);
+    }
+
+    public Page<SubmissionEnvelope> getProjectSubmissionEnvelopes(Project project, Pageable pageable) {
+        Set<SubmissionEnvelope> envelopes = new HashSet<>();
+        envelopes.add(project.getSubmissionEnvelope());
+        Page<Project> projects = this.projectRepository.findByUuid(project.getUuid(), Pageable.unpaged());
+        for (Project projectDocument : projects)
+        {
+            envelopes.addAll(projectDocument.getSubmissionEnvelopes());
```

Comment:
If one of these could add nulls to the collection, then I think it's safer to pre-check that we're not adding null in the first place rather than cleaning up nulls later e.g 

```
projectDocument.getSubmissionEnvelopes().forEach(maybeEnvelope -> Optional.ofNullable(envelope).ifPresent(envelope -> envelopes.add(envelope)))
```

Reply:
If I do it this other way I can log a warning if the project has references to deleted submissions